### PR TITLE
Order DTO and services

### PR DESCRIPTION
**Pull request by IntensioT**
Created order dto, configured mapper for order dto
Added OrderService interface and implementation, validators for order dto

### CODE REVIEW

File: OnlineShop/OnlineShop.BLL/Mappers/MapsterConfig.cs [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    {
+        //TypeAdapterConfig<string, Guid>.NewConfig()
+        //    .MapWith(src => StringToGuidMapper.ToGuid(src));
+
+        //TypeAdapterConfig<Guid, string>.NewConfig()
+        //    .MapWith(src => StringToGuidMapper.ToString(src));
+
+        //TypeAdapterConfig<RegisterRequestDTO, User>.NewConfig()
+        //    .Map(dest => dest.PasswordHash, src => PasswordHelper.HashPassword(src.Password))
+        //    .Map(dest => dest.IsDeleted, src => false);
```
Comment (mebry) [inline OnlineShop/OnlineShop.BLL/Mappers/MapsterConfig.cs]: delete unnecessary code if it is commented out

File: OnlineShop/OnlineShop.BLL/Services/Implementations/OrderService.cs [hunk 2]
```diff
@@ -1,8 +1,109 @@
...
+
+        await _orderRepository.CreateAsync(order, cancellationToken);
+
+        var orderResponseDTO = order.Adapt<OrderResponseDTO>();
+        return orderResponseDTO;
+    }
+
+    public async Task<IEnumerable<OrderResponseDTO>> GetOrdersByUserIdAsync(string userId, CancellationToken cancellationToken = default)
+    {
+        var id = userId.Adapt<Guid>();
```
Comment (mebry) [inline OnlineShop/OnlineShop.BLL/Services/Implementations/OrderService.cs]: Redo it as in the previous pool request

File: OnlineShop/OnlineShop.BLL/Services/Implementations/OrderService.cs [hunk 3]
```diff
@@ -1,8 +1,109 @@
...
+    {
+        var order = await _orderRepository.GetByIdAsync(orderId, cancellationToken);
+        if (order is null)
+        {
+            throw new EntityNotFoundException(nameof(Order), orderId);
+        }
+
+        if (order.IsCancelled)
+        {
+            throw new ValidationException("This order has already been cancelled.");
```
Comment (mebry) [inline OnlineShop/OnlineShop.BLL/Services/Implementations/OrderService.cs]: This is not a validation exception

**Review by mebry — APPROVED** [review]

**Comment by KiSSel3** [comment]
Corrected errors

**Review by mebry — APPROVED** [review]
update the branch with new changes later, which will come from other branches

**Pull request merged**